File: lingting-grpc/src/main/java/live/lingting/framework/grpc/properties/GrpcServerProperties.java
package live.lingting.framework.grpc.properties;

import live.lingting.framework.util.MdcUtils;
import lombok.Getter;
import lombok.Setter;

import java.time.Duration;

/**
 * @author lingting 2023-04-06 15:18
 */
@Getter
@Setter
public class GrpcServerProperties {

	private Integer port;

	private long messageSize = 524288;

	private Duration keepAliveTime = Duration.ofMinutes(30);

	private Duration keepAliveTimeout = Duration.ofSeconds(2);

	private String traceIdKey = MdcUtils.TRACE_ID;

	private int traceOrder = Integer.MIN_VALUE + 100;

	private int exceptionHandlerOrder = Integer.MIN_VALUE + 200;

}

File: lingting-grpc/src/main/kotlin/live/lingting/framework/grpc/properties/GrpcServerProperties.kt
package live.lingting.framework.grpc.properties

import java.time.Duration
import live.lingting.framework.data.DataSize
import live.lingting.framework.util.MdcUtils

/**
 * @author lingting 2023-04-06 15:18
 */
class GrpcServerProperties {
    var port: Int? = null

    /**
     * 是否使用gzip压缩/解压
     */
    var useGzip: Boolean = false

    var messageSize: Long = DataSize.ofMb(1).bytes

    var keepAliveTime: Duration = Duration.ofMinutes(30)

    var keepAliveTimeout: Duration = Duration.ofSeconds(2)

    var traceIdKey: String = MdcUtils.traceIdKey

    var traceOrder: Int = Int.MIN_VALUE

    var exceptionHandlerOrder: Int = Int.MIN_VALUE + 2000

    /**
     * 当上下文在关闭阶段时, 是否拒绝请求
     * @see live.lingting.framework.application.ApplicationHolder.isStop
     */
    var rejectRequestOnStop = true

}
